src/main/java/com/kobylynskyi/graphql/codegen/mapper/AnnotationsMapper.java

```diff
@@ -123,9 +124,16 @@ public List<String> getAnnotations(MappingContext mappingContext, String graphQL
             }
         }
         // 6. Add annotations for resolver arguments
-        if (!Utils.isEmpty(mappingContext.getResolverArgumentAnnotations())
-                && mappingContext.getOperationsName().contains(parentTypeName)) {
-            annotations.addAll(mappingContext.getResolverArgumentAnnotations());
+        if (!Utils.isEmpty(mappingContext.getResolverArgumentAnnotations())) {
+            if (mappingContext.getOperationsName().contains(parentTypeName)
+                || (def instanceof InputValueDefinition
+                && !mappingContext.getDocument().getInputDefinitions()
```

Comment:
This method (`AnnotationsMapper.getAnnotations()`) is executed for **each and every GraphQL entity** (type/enum/interface/union/input/parameter/etc) which makes it an expensive method.

So collecting the names of all inputs should be done once in `MappingContext` and then should be reused here. (Similar caching approach was done for operation names / interface names / union names).

Please create a `Set` of input names in `MappingContext` and then here do `mappingContext.getInputsName().contains(parentTypeName)`.

Reply:
Thank you for the review. 
It is done.